### PR TITLE
Add step markduplicates & allow BAM input for all steps

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#512](https://github.com/nf-core/sarek/pull/512), [#531](https://github.com/nf-core/sarek/pull/531), [#537](https://github.com/nf-core/sarek/pull/537) - Subway map for pipeline
 - [#522](https://github.com/nf-core/sarek/pull/522) - Add QC for vcf files & MultiQC
 - [#533](https://github.com/nf-core/sarek/pull/533) - Add param `--only_paired_variant_calling` to allow skipping of germline variantcalling for paired samples
+- [#536](https://github.com/nf-core/sarek/pull/536) - Add `--step markduplicates` to start from duplicate marking, `--step prepare_recalibration` now ONLY starts at process `BaseRecalibrator` & adding `bam` and `cram` input support for `--step markduplicates,prepare_recalibration,recalibrate`
```
Comment (maxulysse) [inline CHANGELOG.md:26]:
```suggestion
- [#536](https://github.com/nf-core/sarek/pull/536) - Add `--step markduplicates` to start from duplicate marking, `--step prepare_recalibration` now ONLY starts at process `BaseRecalibrator` & adding `bam` and `cram` input support for `--step` `markduplicates`, `prepare_recalibration` and `recalibrate`
```

Comment (maxulysse) [reply CHANGELOG.md:26]: don't want people to get the idea of using multiple steps

Comment (FriederikeHanssen) [reply CHANGELOG.md:26]: oh right, that would not be good

File: conf/test.config [hunk 2]
```diff
@@ -55,10 +55,26 @@ profiles {
     pair {
         params.input               = "${baseDir}/tests/csv/3.0/fastq_pair.csv"
     }
-    prepare_recalibration {
+    markduplicates_bam {
+        params.input               = "${baseDir}/tests/csv/3.0/mapped_single.csv"
```
Comment (maxulysse) [inline conf/test.config:59]: What about mapped_single.csv > mapped_single_bam.csv to be more explicit?

**Comment by maxulysse** [comment]
why not adding `DEEPTOOLS_BAMCOVERAGE` to `nf-core/cram_qc.nf` and control in `conf/modules.config` which qc tools we use?

**Comment by FriederikeHanssen** [comment]
> why not adding `DEEPTOOLS_BAMCOVERAGE` to `nf-core/cram_qc.nf` and control in `conf/modules.config` which qc tools we use?

hm yeah that would work, or I thought also that we could rewrite bam_qc to run the QC tools on cram both and just control SAMTOOLS_BAMTOCRAM whether it is cram input or not.

**Review by marissaDubbelaar — APPROVED** [review]
LGTM, two minor suggestions

File: workflows/sarek.nf [hunk 3]
```diff
@@ -587,11 +670,36 @@ workflow SAREK {
...
+            // ch_cram_variant_calling contains either:
+            // - crams from markduplicates = ch_cram_for_prepare_recalibration if skip BQSR but not started from step recalibration
+            cram_variant_calling = Channel.empty().mix(ch_cram_for_prepare_recalibration)
+        }
     }
 
-    if (params.step == 'variant_calling') cram_variant_calling = ch_input_sample
+    if (params.step == 'variant_calling') {
+
+        ch_input_sample.view()
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply workflows/sarek.nf:688]: no, will remove

File: workflows/sarek.nf [hunk 4]
```diff
@@ -49,6 +49,7 @@ else {
     log.warn "No samplesheet specified, attempting to restart from csv files present in ${params.outdir}"
     switch (params.step) {
         case 'mapping': exit 1, "Can't start with step $params.step without samplesheet"
+        //case 'markduplicates': csv_file = file("${params.outdir}/preprocessing/csv/markduplicates_no_table.csv", checkIfExists: true); break
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply workflows/sarek.nf:52]: yes and no, unsure yet if we can implement automatic restart as we had before. But this also requires more upstream fixes, so as a reminder i'd like to keep it here for now

**Review by ggabernet — APPROVED** [review]
Hey Looks good to me! I've just thought that if you changed that the step `prepare_recalibration` works maybe that should also be added to the docs, but maybe you plan to add all docs in a separate PR, that's also fine :)

**Comment by FriederikeHanssen** [comment]
> Hey Looks good to me! I've just thought that if you changed that the step `prepare_recalibration` works maybe that should also be added to the docs, but maybe you plan to add all docs in a separate PR, that's also fine :)

Yes I am planning to do everything in the docs at once, but it is probably a good idea to at least put a note into the docs so i won't loose track.

Thanks a lot for reviewing  🙏

**Pull request merged**